src/library_dylink.js

```diff
@@ -283,7 +283,7 @@ var LibraryDylink = {
       assert(dylinkSection.length != 0, 'need dylink section');
       binary = new Int8Array(dylinkSection[0]);
     } else {
-      var int32View = new Uint32Array(new Uint8Array(binary.subarray(0, 24)).buffer);
+      var int32View = new Uint32Array(new Uint8Array(binary.subarray(0, 4)).buffer);
```

Comment:
This line seems to do unnecessary wrapping. If it's meant to be used in read-only fashion (which it seems to be), then

```suggestion
      var int32View = new Uint32Array(binary.buffer, 0, 1);
```

should be enough.

Reply:
I tried all kind of way to do basically what you say here... but it always seemed to fail when binary was a view into an buffer than was not at the start.     `new Uint32Array(binary.buffer, 0, 1);` seemed to always give me the first 32bits of the underlying buffer.. but I want the first 32bits of the view I'm being passed.

The time this happens in emscripten is where many files are packed into single buffer,  and `binary` is a Uint8Array slice/view into that buffer at some offset.   (The offset might not even be 4-byte aligned I think).

At least that was what seems to be happening.   this really confused me a long time ..and I can't say I understand why this convoluted expression fixes it.

Reply:
Ah, I see, that makes sense if you want to support arbitrary views. I guess in that case the only simplification I can offer is

```suggestion
      var int32View = new Uint32Array(binary.slice(0, 4).buffer);
```

Reply:
I just expected that, since this is a beginning of the module, it should always be beginning of the buffer as well.